### ISSUE TITLE
bug report: app crashes when removing actions

### ISSUE THREAD

**Issue by Efreak** [Nain57/Smart-AutoClicker#108]
I've got a scenario with multiple actions defined; currently i want an action repeated a certain number of times with a delay between, so this is implemented as follows, with a mistake. Attempting to delete that last action, the app crashes.

![Screen_Recording_20220311-123551_FastHub-Libre mp4001](https://user-images.githubusercontent.com/749751/157974986-2ed24e61-4fdf-445a-8573-97545f28a96b.png)

Edit: I managed to get a screen recording: [Screen_Recording_20220311-141615_Skill Quest.cut.mp4](https://gist.github.com/Efreak/0b397e3b5ffd5d9deecbdbf37eda318d/raw/c8a47f415abff9fe5a8c9d52f643afb4128022ee/Screen_Recording_20220311-141615_Skill%2520Quest.cut.mp4)

<details>
<summary>App version: 1.2.4 (via fdroid)</summary>

- Android version: 11
- Device manufacturer: samsung
- Device model: SM-T860
</details>

<details>
<summary>logcat (filtered to include only autoclicker pid)</summary>

Function used to filter logcat:
```
#filterlogcatpidsbyname packagename [logcat.txt]
function filterlogcatpidsbyname(){
	logfile=$2
	logfile=${logfile:=logcat.txt}
	pidlist=$(grep -E "?Process:.*$1.*PID:" "$logfile" | col3 |tr '\n' '|' | sed -E 's/\|$//')
	grep -E "^[0-9-]+\s*[:\.0-9]+\s*?($pidlist)" "$logfile"
}
filterlogcatpidsbyname buzbuz. | termux-clipboard-set
```
There's only one crash here; the rest are in [buzbuz.logcat.txt](https://gist.github.com/Efreak/0b397e3b5ffd5d9deecbdbf37eda318d#file-buzbuz-logcat-txt), as this would otherwise be too long (the other crashes are pretty much the same anyways)

```
03-11 11:36:24.261  3570  3570 E AndroidRuntime: FATAL EXCEPTION: main
03-11 11:36:24.261  3570  3570 E AndroidRuntime: Process: com.buzbuz.smartautoclicker, PID: 3570
03-11 11:36:24.261  3570  3570 E AndroidRuntime: java.lang.IndexOutOfBoundsException: Index: 4, Size: 4
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at java.util.ArrayList.get(ArrayList.java:437)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at v1.h.c(:1)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$t.j(:24)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager$c.c(:3)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.f1(Unknown Source:0)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.S0(:2)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.l0(:17)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.s(:2)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onMeasure(:2)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.cardview.widget.CardView.onMeasure(Unknown Source:0)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.core.widget.NestedScrollView.measureChildWithMargins(Unknown Source:38)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.core.widget.NestedScrollView.onMeasure(Unknown Source:0)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.appcompat.widget.AlertDialogLayout.onMeasure(:1)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(Unknown Source:154)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7951)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:1218)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.View.measure(View.java:27137)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:4257)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3701)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2669)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10092)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1010)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:809)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:744)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:247)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8656)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
03-11 11:36:24.261  3570  3570 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

</details>

**Comment by Nain57** [comment]
Thanks for this very detailed report.
This crash will be addressed for the next version.

**Comment by Nain57** [comment]
Fixed in 1.3.0. Should be available soon on F-Droid
